### PR TITLE
Fix Tenant wise internal OAuth2 scopes not getting populated

### PR DESCRIPTION
**Pull request by DMHP**
## Purpose
Fix https://github.com/wso2/product-is/issues/10505

### CODE REVIEW

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5100/dao/OAuth2ScopeDAO.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+     */
+    public void addScope(Scope scope, int tenantID) throws MigrationClientException {
+
+        if (scope == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Scope is not defined");
+            }
+
+            throw new MigrationClientException("Error occured while migrating the internal OAuth2 scopes for the tenant " +
+                    ":"+tenantID+". Scope Name is not specified.");
```
Comment (Buddhimah) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5100/dao/OAuth2ScopeDAO.java:57]:
```suggestion
                    ": " + tenantID + ". Scope Name is not specified.");
```

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5100/dao/OAuth2ScopeDAO.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        } catch (SQLException e) {
+            String msg = "Error occurred while creating scope :" + scope.getName();
+            throw new MigrationClientException(msg, e);
+        }
+    }
+
+    private void addScope(Scope scope, Connection conn, int tenantID)
+            throws SQLException {
+
+        //Adding the scope
```
Comment (Buddhimah) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5100/dao/OAuth2ScopeDAO.java:82]:
```suggestion
        //Adding the scope.
```

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5100/dao/OAuth2ScopeDAO.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                ps.execute();
+
+                try (ResultSet rs = ps.getGeneratedKeys()) {
+                    if (rs.next()) {
+                        scopeID = rs.getInt(1);
+                    }
+                }
+            }
+
+            // some JDBC Drivers returns this in the result, some don't
```
Comment (Buddhimah) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5100/dao/OAuth2ScopeDAO.java:102]:
```suggestion
            // Some JDBC Drivers returns this in the result, some don't.
```

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5100/dao/OAuth2ScopeDAO.java [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+            if (scopeID == 0) {
+                if (log.isDebugEnabled()) {
+                    log.debug("JDBC Driver did not return the scope id, executing Select operation");
```
Comment (Buddhimah) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5100/dao/OAuth2ScopeDAO.java:105]:
```suggestion
                    log.debug("JDBC Driver did not return the scope id, executing Select operation.");
```

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5100/util/OAuth2Util.java [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+                if (log.isDebugEnabled()) {
+                    log.debug("OAuth scopes are loaded for the tenant : " + tenantId);
+                }
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("OAuth scopes are already loaded");
+                }
+            }
+        } catch (MigrationClientException e) {
+            log.error("Error while registering OAuth scopes with permissions bindings", e);
```
Comment (Buddhimah) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5100/util/OAuth2Util.java:81]:
```suggestion
            log.error("Error while registering OAuth scopes with permissions bindings.", e);
```

Comment (Buddhimah) [reply components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5100/util/OAuth2Util.java:81]: Check other places as well

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5100/dao/OAuth2ScopeDAO.java [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+            if (log.isDebugEnabled()) {
+                log.debug("Scope is not defined");
+            }
+
+            throw new MigrationClientException("Error occured while migrating the internal OAuth2 scopes for the tenant " +
+                    ":"+tenantID+". Scope Name is not specified.");
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Adding scope :" + scope.getName());
```
Comment (Buddhimah) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5100/dao/OAuth2ScopeDAO.java:61]:
```suggestion
            log.debug("Adding scope: " + scope.getName());
```

Comment (Buddhimah) [reply components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5100/dao/OAuth2ScopeDAO.java:61]: Check other places as well

**Review by IsuraD — APPROVED** [review]

**Pull request merged**